### PR TITLE
Create Blog “set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise”

### PR DESCRIPTION
**Pull request by GuopingJia**
Automatically generated by Netlify CMS

### CODE REVIEW

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+### Overview
+
+In [HPE GreenLake for Private Cloud Enterprise: Containers](https://www.hpe.com/us/en/greenlake/containers.html), customers can create services with the type of *NodePort* for their workloads deployed in K8s clusters using the label *hpecp.hpe.com/hpecp-internal-gateway=true*. The services will be automatically exposed to a container platform gateway host with assigned ports. The deployed workloads will become accessible externally using the gateway host name and the assigned ports as access URLs. Different from various public cloud providers, such as *GCP*, *AWS* and *Microsoft Azure*, HPE GreenLake for Containers doesn’t support network load balancers. When you create services with the type of *LoadBalancer*, they will remain in the *'pending'* state indefinitely when created. This blog post will show you how to use MetalLB to provide load balancing services for K8s clusters in HPE GreenLake for Containers. It provides customers with the flexibility to configure custom load balancers for their deployed workloads in HPE GreenLake for Private Cloud Enterprise.
```
Comment (Didier-Lalli) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:19]: Isn't it a problem to clearly say, we don't support network load balancers? Seems like you are exposing a bug when you describe the pending state. I would rephrase in a more positive way.

Also you use HPE GreenLake for Contain twice although you used HPE GreenLake for Private Cloud Enterprise: Containers. As always we are very confusing with our naming convention :-)

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+clusterrolebinding.rbac.authorization.k8s.io/metallb-system:controller created
+clusterrolebinding.rbac.authorization.k8s.io/metallb-system:speaker created
+secret/webhook-server-cert created
+service/webhook-service created
+deployment.apps/controller created
+daemonset.apps/speaker created
+validatingwebhookconfiguration.admissionregistration.k8s.io/metallb-webhook-configuration created
+```
+
+The above command installs the latest MetalLB *v0.13.10* to the K8s cluster. It first creates the namespace *metallb-system*, sets up the role-based access control (*RBAC*), creates a list of customer resource definitions (CRDs) andfinally deploys a list of pods and services.
```
Comment (Didier-Lalli) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:69]: Typo: andfinally, missing a space.

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+service/cfe-nginx-app   LoadBalancer   10.98.244.64   172.16.17.250   80:31631/TCP   3m22s
+
+NAME                            READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/cfe-nginx-app   1/1     1            1           3m21s
+
+NAME                                       DESIRED   CURRENT   READY   AGE
+replicaset.apps/cfe-nginx-app-66cb4f5bbf   1         1         1       3m22s
+```
+
+T﻿o verify the deployed Nginx application is working, l﻿aunch your web browser, type the string "http://" followed by the IP address *'172.16.17.250'* that's assigned t﻿o the Nginx service, then press the *Enter* key. You should see the following texts showing in the browser:
```
Comment (Didier-Lalli) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:222]: I'd just say open http://172.16.17.250 with your web browser. You are using double quotes, then simple quotes. Also it's missing a space before that's assigned.

Don't need to tell them to press enter IMO.

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+l2advertisement.metallb.io/example created
+
+$ kubectl get L2Advertisement -n metallb-system
+NAME      IPADDRESSPOOLS   IPADDRESSPOOL SELECTORS   INTERFACES
+example   ["cfe-pool"]               
+```
+
+### D﻿eploy Nginx app as the service type *LoadBalancer*
+
+As a sample web application, the *Nginx* with the service type of *LoadBalancer* will be deployed to the K8s cluster using the following YAML manifest file:
```
Comment (Didier-Lalli) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:149]: Maybe point out the fact that you have a custom index.html since you'll be showing it in the next step.

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+ 
+![](/img/web-nginx-app.png)
+
+
+
+
+### Summary
+
+This blog post describes the detailed process used to deploy and set up MetalLB to support customers to configure load balancers for the K8s clusters in HPE GreenLake for Private Cloud Enterprise. By deploying load balancing configuration, it allows customers to easily work together with Kubernetes Ingress for different traffic routing support of their deployed workloads in the K8s clusters. It also supports deploying applications by passing the customer certificates through their own authority. It unblocks a list of potential use cases and enhances HPE GreenLake product with additional flexibilities.
+
```
Comment (Didier-Lalli) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:234]: I find this sentence "By deploying load balancing configuration, it allows customers to easily work together with Kubernetes Ingress for different traffic routing support of their deployed workloads in the K8s clusters" a bit long and unclear. Maybe you could rephrase.

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+NAME                            READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/cfe-nginx-app   1/1     1            1           3m21s
+
+NAME                                       DESIRED   CURRENT   READY   AGE
+replicaset.apps/cfe-nginx-app-66cb4f5bbf   1         1         1       3m22s
+```
+
+T﻿o verify the deployed Nginx application is working, l﻿aunch your web browser, type the string "http://" followed by the IP address *'172.16.17.250'* that's assigned t﻿o the Nginx service, then press the *Enter* key. You should see the following texts showing in the browser:
+
+*﻿Hi, this is the sample <font color=blue>Nginx App</font> deployed as the Load Balancer service type !*
```
Comment (Didier-Lalli) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:224]: Since you are showing a screen capture, maybe you don't need to spell out what's the text going to be.

**Review by Didier-Lalli — CHANGES_REQUESTED** [review]

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md (hunk 1 above)
Comment (drensing) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:19]: I recommend that this first sentence reads as follows:
"Using HPE GreenLake for Containers, customers can create services with the..."

I also recommend breaking this up into two paragraphs, starting the second paragraph with "Different from various public cloud providers, such as GCP..."

Where it says "This blog post will show you how to use MetalLB to provide load balancing services for K8s clusters in HPE GreenLake for Containers"... do you actually mean "using HPE GreenLake for Containers"? I'm wondering about the word "in" and whether a different term should be used.

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+
+### Overview
+
+In [HPE GreenLake for Private Cloud Enterprise: Containers](https://www.hpe.com/us/en/greenlake/containers.html), customers can create services with the type of *NodePort* for their workloads deployed in K8s clusters using the label *hpecp.hpe.com/hpecp-internal-gateway=true*. The services will be automatically exposed to a container platform gateway host with assigned ports. The deployed workloads will become accessible externally using the gateway host name and the assigned ports as access URLs. Different from various public cloud providers, such as *GCP*, *AWS* and *Microsoft Azure*, HPE GreenLake for Containers doesn’t support network load balancers. When you create services with the type of *LoadBalancer*, they will remain in the *'pending'* state indefinitely when created. This blog post will show you how to use MetalLB to provide load balancing services for K8s clusters in HPE GreenLake for Containers. It provides customers with the flexibility to configure custom load balancers for their deployed workloads in HPE GreenLake for Private Cloud Enterprise.
+
+### Prerequisites
+
+Before starting, make sure you have the following requirements:
```
Comment (drensing) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:23]: Change to either:
"Before starting, make sure you have the following:"    

or

"Before starting, make sure you meet the following requirements:"

(you can't actually "have" requirements in this scenario. You can "have requirements" in other scenarios, but not here...)

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+disable: false
+tags:
+  - Kubernetes
+  - MetalLB
+  - Load Balancer
+  - HPE GreenLake
+  - HPE GreenLake for Private Cloud Enterprise
+  - HPE GreenLake for Containers
+---
+This blog post describes how to set up the load balancer using [MetalLB](https://metallb.universe.tf/) for a K8s cluster in [HPE GreenLake for Private Cloud Enterprise](https://www.hpe.com/us/en/greenlake/private-cloud-enterprise.html). It allows customers to configure load balancing services for their workloads deployed in the K8s clusters.
```
Comment (drensing) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:15]: In the first instance where you call out K8s, first spell out Kubernetes and place (K8s) in parentheses afterwards...
"Kubernetes (K8s)." Then, be consistent on whether you're going to shorten it throughout the blog post using one or the other  consistently.

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md (hunk 1 above)
Comment (drensing) [reply content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:19]: I'm pretty sure the HPE GreenLake for Containers is a separate entity... I found it on this website: https://www.hpe.com/us/en/greenlake/containers.html

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+
+Before starting, make sure you have the following requirements:
+
+* A K8s cluster, being provisioned in HPE GreenLake for Private Cloud Enterprise
+* The kubectl CLI tool, together with the kubeconfig file for accessing the K8s cluster
+* A range of virtual IP addresses. Those IP addresses should not be used in any existing K8s clusters. They will be assigned to the load balancer services. 
+
+### D﻿eploy MetalLB for load balancing
+
+[MetalLB](https://metallb.universe.tf/) is a software solution that provides a network load balancer implementation for Kubernetes clusters using standard routing protocols. By installing MetalLB, it will support the LoadBalancer services within the Kubernetes clusters.
```
Comment (drensing) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:31]: Here you start referring to Kubernetes versus K8s. I'd just be consistent and pick one, but make sure, if you use K8s, that you've defined it in the first instance.

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+
+* A K8s cluster, being provisioned in HPE GreenLake for Private Cloud Enterprise
+* The kubectl CLI tool, together with the kubeconfig file for accessing the K8s cluster
+* A range of virtual IP addresses. Those IP addresses should not be used in any existing K8s clusters. They will be assigned to the load balancer services. 
+
+### D﻿eploy MetalLB for load balancing
+
+[MetalLB](https://metallb.universe.tf/) is a software solution that provides a network load balancer implementation for Kubernetes clusters using standard routing protocols. By installing MetalLB, it will support the LoadBalancer services within the Kubernetes clusters.  
+
+This section describes the detailed steps to deploy MetalLB and configure it to support the *LoadBalancer* services in the Kubernetes clusters.
```
Comment (drensing) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:33]: Same note here re: Kubernetes

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+
+#### 1 Deploy MetalLB:
+
+The MetalLB can be deployed by applying the following YAML manifest file:
```
Comment (drensing) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:37]: Not sure you really need "The" prior to MetalLB

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md (hunk 2 above)
Comment (drensing) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:69]: In the last line, you need a space between "and" and "finally"

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+secret/webhook-server-cert created
+service/webhook-service created
+deployment.apps/controller created
+daemonset.apps/speaker created
+validatingwebhookconfiguration.admissionregistration.k8s.io/metallb-webhook-configuration created
+```
+
+The above command installs the latest MetalLB *v0.13.10* to the K8s cluster. It first creates the namespace *metallb-system*, sets up the role-based access control (*RBAC*), creates a list of customer resource definitions (CRDs) andfinally deploys a list of pods and services. 
+
+You can check and confirm all pods and services are deployed successful:
```
Comment (drensing) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:71]: Pretty sure you mean "successfully" at the end here...

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md (hunk 3 above)
Comment (drensing) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:222]: Instead of saying "You should see the following texts showing in the browser:", I'd recommend rephrasing this to:
"The following should now show in your browser:"

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+*﻿Hi, this is the sample <font color=blue>Nginx App</font> deployed as the Load Balancer service type !*
+ 
+![](/img/web-nginx-app.png)
+
+
+
+
+### Summary
+
+This blog post describes the detailed process used to deploy and set up MetalLB to support customers to configure load balancers for the K8s clusters in HPE GreenLake for Private Cloud Enterprise. By deploying load balancing configuration, it allows customers to easily work together with Kubernetes Ingress for different traffic routing support of their deployed workloads in the K8s clusters. It also supports deploying applications by passing the customer certificates through their own authority. It unblocks a list of potential use cases and enhances HPE GreenLake product with additional flexibilities.
```
Comment (drensing) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:233]: Change the last sentence to read "It unblocks a list of potential use cases and enhances HPE GreenLake by providing additional flexibility."

Also, consider adding some sort of "call to action"... maybe suggest that they read up more on a different website or suggest they keep coming back to the HPE Developer blog to learn more about HPE GreenLake for Private Cloud Enterprise.

**Review by drensing — COMMENTED** [review]
Hi, Jia! Great to see you again. Thank you for your blog post! Your posts are great and I see that you've already remembered some of the things I've recommended in previous editorial cycles.

All the best,
Dale

**Comment by GuopingJia** [comment]
> Hi, Jia! Great to see you again. Thank you for your blog post! Your posts are great and I see that you've already remembered some of the things I've recommended in previous editorial cycles.
> 
> All the best, Dale

Thanks Dale for your review comments. I finally get some time today and I will walk through your comments and address them, together with Didier's ones, in the new patch.  

Guoping

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md (hunk 1 above)
Comment (GuopingJia) [reply content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:19]: > Isn't it a problem to clearly say, we don't support network load balancers? Seems like you are exposing a bug when you describe the pending state. I would rephrase in a more positive way.
> 
> Also you use HPE GreenLake for Contain twice although you used HPE GreenLake for Private Cloud Enterprise: Containers. As always we are very confusing with our naming convention :-)

This is a very good comment. I will rephrase the load balancer support - see below. I am also deleting the pending state in the loadbalancer service. It's a consequence of missing load balancer support, it's not a bug.

"HPE GreenLake for Private Cloud Enterprise doesn’t support network load balancers by default. As cluster administrators, after you create K8s cluster, you can integrate cluster with any load balancers in place to support K8s services of type *LoadBalancer*. "

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md (hunk 8 above)
Comment (GuopingJia) [reply content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:15]: A good comment. I like it !

**Comment by GuopingJia** [comment]
Hi, Didier, Dale, 

The new patch should address all your comments. Please let me know if there is still anything you want me to change / discuss. 

Didier, apologiz for not being able to get back to you in time for your review comments !

Guoping

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 14]
```diff
@@ -0,0 +1,236 @@
...
+
+Different from various public cloud providers, such as *GCP*, *AWS* and *Microsoft Azure*, HPE GreenLake for Private Cloud Enterprise doesn’t support network load balancers by default. As cluster administrators, after you create K8s cluster, you can integrate cluster with any load balancers in place to support K8s services of type *LoadBalancer*. This blog post will show you how to use MetalLB to provide load balancing services for K8s clusters in HPE GreenLake for Private Cloud Enterprise. It provides customers with the flexibility to configure custom load balancers for their deployed workloads in HPE GreenLake for Private Cloud Enterprise.
+
+### Prerequisites
+
+Before starting, make sure you meet the following requirements:
```
Comment (Didier-Lalli) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:26]: If you add the following, the bullet list will use the same styl as the rest of the text

<style>
li {
   font-size: 27px;
   line-height: 33px;
   max-width: none;
}
</style>

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 15]
```diff
@@ -0,0 +1,236 @@
...
+* The kubectl CLI tool, together with the kubeconfig file for accessing the K8s cluster
+* A range of virtual IP addresses. Those IP addresses should not be used in any existing K8s clusters. They will be assigned to the load balancer services. 
+
+### D﻿eploy MetalLB for load balancing
+
+[MetalLB](https://metallb.universe.tf/) is a software solution that provides a network load balancer implementation for Kubernetes clusters using standard routing protocols. By installing MetalLB, it will support the LoadBalancer services within the Kubernetes clusters.  
+
+This section describes the detailed steps to deploy MetalLB and configure it to support the *LoadBalancer* services in the Kubernetes clusters.
+
+#### 1 Deploy MetalLB:
```
Comment (Didier-Lalli) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:38]: I'd use 1. or 1/ 
Also this one has a terminating : but not the other ones. I suggest removing the :

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 16]
```diff
@@ -0,0 +1,236 @@
...
+daemonset.apps/speaker   6         6         4       6            4           kubernetes.io/os=linux   37s
+
+NAME                         READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/controller   0/1     1            0           38s
+
+NAME                                   DESIRED   CURRENT   READY   AGE
+replicaset.apps/controller-7967ffcf8   1         1         0       38s
+```
+
+#### 2 Define a range of IP addresses
```
Comment (Didier-Lalli) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:100]: Same as 1

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 17]
```diff
@@ -0,0 +1,236 @@
...
+ipaddresspool.metallb.io/cfe-pool created
+
+$ kubectl get IPAddressPool -n metallb-system
+NAME       AUTO ASSIGN   AVOID BUGGY IPS   ADDRESSES
+cfe-pool   true          false             ["172.16.17.250-172.16.17.254"]
+```
+
+The above command allocates the IP pool that has the IP range 172.16.17.250-172.16.17.254. The IP addresses in the *IPAddressPool* can be defined by *CIDR* and *IPV6* addresses as well.
+
+#### 3 Announce the service IP addresses
```
Comment (Didier-Lalli) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:127]: Same as 1 and 2

**Review by Didier-Lalli — CHANGES_REQUESTED** [review]

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md (hunk 14 above)
Comment (GuopingJia) [reply content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:26]: Thank you for pointing me this. I've added this, but updated the values to font-size / line-height:

<style> li { font-size: 100%; line-height: 23px; max-width: none; } </style>

Otherwise, the font / line look too big. Not sure if I got it correctly.

**Comment by GuopingJia** [comment]
Didier, 

The new patch has addressed your comments today. Please review it. 

Guoping

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md (hunk 14 above)
Comment (Didier-Lalli) [reply content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:26]: If it's ok with you, it'ok for me too :-)

Comment (GuopingJia) [reply content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:26]: I like this markdown css. I will definitely reuse it in my coming blog post :-)

**Review by Didier-Lalli — APPROVED** [review]

File: content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md [hunk 18]
```diff
@@ -0,0 +1,236 @@
...
+```
+
+T﻿o verify the deployed Nginx application is working, l﻿aunch your web browser a﻿nd open *http://172.16.17.250*. The following should now show in your browser:
+
+![](/img/web-nginx-app.png)
+
+### Summary
+
+This blog post describes the detailed process used to deploy and set up MetalLB to support customers to configure load balancers for K8s clusters in HPE GreenLake for Private Cloud Enterprise. By deploying load balancing configuration, it provides an externally accessible IP address that sends traffic to the deployed workload. It also allows customers to use [Kubernetes Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/), in place of *Service*, for different traffic routing support of their deployed workloads in K8s clusters. T
+his supports deploying applications by passing the customer certificates through their own authority. It unblocks a list of potential use cases and enhances HPE GreenLake by providing additional flexibilities.
```
Comment (Didier-Lalli) [inline content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md:234]: There is a problem with this, between the T and the h.
I'm good with the rwst.

Comment (GuopingJia) [reply content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md]: Sorry, it's my fault. It's been fixed.
